doc: Chapter 01：Installing WinGet (#191)
OpenINF/.github: doc/collections/_pages/content/handbook/setup--windows/chapter-1/Readme.md

# Chapter I: Install winget[^1]

&nbsp;&nbsp;

> This is not as simple as it may sound

&nbsp;&nbsp;

> https://aka.ms/appinst

Author: ```suggestion
https://aka.ms/appinst

As a URL shorten-er & forwards to the GitHub repo below

https://github.com/microsoft/winget-cli
```

> # Chapter I: Install winget[^1]

Author: ```suggestion

<!-- Begin GitHub-flavored Markdown -->

# Chapter I: Install Winget[^1]
```

# Chapter I: Install winget[^1]

&nbsp;&nbsp;

> This is not as simple as it may sound

&nbsp;&nbsp;

https://aka.ms/appinst

As a URL shorten-er & forwards to the GitHub repo below

https://github.com/microsoft/winget-cli

&nbsp;&nbsp;

### Setup

&nbsp;&nbsp;

You will need the following setup to successfully install the MSIX bundle:

&nbsp;&nbsp;

- Windows 10 SDK (version 1809 or above)
- Converted x64 and x86 MSIX packages

&nbsp;&nbsp;

https://apps.microsoft.com/store/detail/app-installer/9NBLGGH4NNS1

&nbsp;&nbsp;

> Microsoft App Installer for Windows 10 makes sideloading Windows 10 apps easy:
  Just double-click the app package, and you won't have to run PowerShell to install apps. App Installer presents the package information like app name, publisher, version, display logo, and the capabilities requested by the app. Get right into the app, no hassles--and if installation doesn't work, the error messages were designed to help you fix the problem. Windows Package Manager is supported through App Installer starting on Windows 10 1809. This application is currently only available for desktop PCs.

&nbsp;&nbsp;

> // NOTE:
> 
> &nbsp;&nbsp;
> 
> > Currently, the Windows server SKUs don’t ship with the Microsoft Store client enabled. This creates challenges for updating applications delivered by the Microsoft Store. As the Windows Package Manager ships as a part of the Desktop App Installer, and that is also not generally default on server SKUs [&hellip;]

Author: ```suggestion
///NOTE/// Currently, the Windows server SKUs don’t ship with the Microsoft Store client enabled. This creates challenges for updating applications delivered by the Microsoft Store. As the Windows Package Manager ships as a part of the Desktop App Installer, and that is also not generally default on server SKUs [&hellip;]

<!-- TODO: References? -->
```

as there will be errors as shown below.

&nbsp;&nbsp;

> Windows cannot install package Microsoft.DesktopAppInstaller_1.19.3132.0_x64\_\_8wekyb3d8bbwe because this package
> depends on a framework that could not be found. Provide the framework "Microsoft.UI.Xaml.2.7" published by
> "CN=Microsoft Corporation, O=Microsoft Corporation, L=Redmond, S=Washington, C=US", with neutral or x64 processor
> architecture and minimum version 7.2109.13004.0, along with this package to install. The frameworks with name
> "Microsoft.UI.Xaml.2.7" currently installed are: {}

&nbsp;&nbsp;

[^1]: https://learn.microsoft.com/en-us/windows/package-manager/winget/

&nbsp;&nbsp;

`MakeAppx.exe` is a tool available in the Windows 10 SDK that allows for packaging and bundling of MSIX packages.

&nbsp;&nbsp;

`MakeAppx.exe` can be used to extract the file contents of a Windows 10 app package or bundle. It also encrypts and decrypts app packages and bundles.

&nbsp;&nbsp;

After the Windows 10 SDK is installed, depending on where it was installed,
`MakeAppx.exe` is usually found here:

&nbsp;&nbsp;

`C:\Program Files (x86)\Windows Kits\10\bin\10.0.22621.0\x64\makeappx.exe`

&nbsp;&nbsp;

> Add this directory to your `PATH`

Author: This is not as straightforward as it may sound, but would like to include something in PowerShell

    Write-Information "The file $($package.filename) already exist in the repo. Skip download"
}

# Install the Runtime
add-AppPackage (Join-Path $RepoPath -ChildPath $package.filename)
```

&nbsp;&nbsp;

```powershell
# Install-Winget Version v1.0.11692
# Derek Lewis 2022 Derivative Work originally by Andreas Nick 2021

# From github
$WinGet_Link = 'https://github.com/microsoft/winget-cli/releases/download/v1.4.3132-preview/Microsoft.DesktopAppInstaller_8wekyb3d8bbwe.msixbundle'
$WinGet_Name = 'Microsoft.DesktopAppInstaller_8wekyb3d8bbwe.msixbundle'

$RepoName = 'AppPackages'
$RepoLokation = $env:Temp
$Packagename = 'Winget'

$RepoPath = Join-Path $RepoLokation -ChildPath $RepoName
$RepoPath = Join-Path $RepoPath -ChildPath $Packagename

if(-not (Test-Path $RepoPath )) {
    New-Item $RepoPath -ItemType Directory -Force
}

if(-not (Test-Path (Join-Path $RepoPath -ChildPath  $WinGet_Name ))) {
    Invoke-WebRequest -Uri $WinGet_Link -OutFile (Join-Path $RepoPath -ChildPath $WinGet_Name )
} else  {
    Write-Information "The file $WinGet_Name already exist in the repo. Skip download"
}

#Install the Package
Add-AppPackage (Join-Path $RepoPath -ChildPath $WinGet_Name)
```

---

> # Acknowledgements:

Author: ```suggestion
## Acknowledgements

&nbsp;&nbsp;
```

<!-- Begin GitHub-Flavored Markdown (GFM) -->

# Chapter I: Install Winget[^1]

&nbsp;&nbsp;

> This is not as simple as it may sound

&nbsp;&nbsp;

https://aka.ms/appinst

As a URL shorten-er & forwards to the GitHub repo below

https://github.com/microsoft/winget-cli

&nbsp;&nbsp;

### Setup

&nbsp;&nbsp;

You will need the following setup to successfully install the MSIX bundle:

&nbsp;&nbsp;

- Windows 10 SDK (version 1809 or above)
- Converted x64 and x86 MSIX packages

&nbsp;&nbsp;

https://apps.microsoft.com/store/detail/app-installer/9NBLGGH4NNS1

&nbsp;&nbsp;

> Microsoft App Installer for Windows 10 makes sideloading Windows 10 apps easy:
  Just double-click the app package, and you won't have to run PowerShell to install apps. App Installer presents the package information like app name, publisher, version, display logo, and the capabilities requested by the app. Get right into the app, no hassles--and if installation doesn't work, the error messages were designed to help you fix the problem. Windows Package Manager is supported through App Installer starting on Windows 10 1809. This application is currently only available for desktop PCs.

&nbsp;&nbsp;

> ///NOTE/// Currently, the Windows server SKUs don’t ship with the Microsoft Store client enabled. This creates challenges for updating applications delivered by the Microsoft Store. As the Windows Package Manager ships as a part of the Desktop App Installer, and that is also not generally default on server SKUs [&hellip;]

Author: ```suggestion
///NOTE///
> Currently, the Windows server SKUs don’t ship with the Microsoft Store client
> enabled. This creates challenges for updating applications delivered by the
> Microsoft Store. As the Windows Package Manager ships as a part of the Desktop
> App Installer, and that is also not generally default on server SKUs
> [&hellip;]
>
> &mdash;https://
```

> <!-- Begin GitHub-Flavored Markdown (GFM) -->

Author: ```suggestion
<!-- Begin GitHub-Flavored Markdown (GFM)

See: https://docs.github.com/get-started/writing-on-github
Spec: https://github.github.com/gfm/
-->
```
